### PR TITLE
fix: remove unused feature flags

### PR DESCRIPTION
**Pull request by just-toby**
## Description

remove dead/unused feature flags

## test plan
automated tests and eyeballs

### CODE REVIEW

File: cypress/e2e/swap/settings.test.ts [hunk 1]
```diff
@@ -24,6 +24,6 @@ describe('Swap settings', () => {
     cy.contains('Transaction deadline').should('exist')
     cy.contains('UniswapX').should('exist')
     cy.contains('Local routing').should('exist')
-    cy.get(getTestSelector('mobile-settings-scrim')).click()
+    cy.get(getTestSelector('mobile-settings-scrim')).click({ force: true })
```
Comment (cbachmeier) [inline cypress/e2e/swap/settings.test.ts:27]: qq: why was this change needed?

File: src/state/routing/slice.ts [hunk 2]
```diff
@@ -146,8 +138,7 @@ export const routingApi = createApi({
               tokenOut: tokenOutAddress,
               amount,
               type,
-              // if forceUniswapXOn is not ON, then use the backend's default value
-              useUniswapX: forceUniswapXOn || undefined,
+              useUniswapX: true,
```
Comment (cbachmeier) [inline src/state/routing/slice.ts:141]: I think this should be undefined here based on the previous comment? `if forceUniswapXOn is not ON, then use the backend's default value` tagging @tinaszheng

File: cypress/e2e/swap/settings.test.ts (hunk 1 above)
Comment (just-toby) [reply cypress/e2e/swap/settings.test.ts:27]: the test was failing for me without this :/ i don't think it's related to my changes though

File: src/state/routing/slice.ts (hunk 2 above)
Comment (just-toby) [reply src/state/routing/slice.ts]: yeah i agree, we should just remove this

**Review by cbachmeier — APPROVED** [review]
